### PR TITLE
[G7T5-34] Create python-app.yml for CI

### PR DESCRIPTION
**Pull request by quinncheong**
# Description

Create python-app.yml for CI

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] Tried to push multiple configurations of the yml file

### CODE REVIEW

File: .github/workflows/python-app.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    paths: ['backend/**', '.github/**']
+  pull_request:
+    branches: [ "master" ]
+    paths: ['backend/**', '.github/**']
+
+permissions:
+  contents: read
+
+env:
+  DAY_OF_WEEK: Monday
```
Comment (jonathantan1425) [inline .github/workflows/python-app.yml:18]: What's this env var for?

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  DB_SERVER: localhost
+  DB_PORT: 3306
+  DB_USER: test
+  DB_PASSWORD: test
+  DATABASE: test
```
Comment (jonathantan1425) [inline .github/workflows/python-app.yml:23]: Will these dummy env vars affect testing? We can assign env vars in github actions separately right?

File: .github/workflows/python-app.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest
```
Comment (jonathantan1425) [inline .github/workflows/python-app.yml:43]: Can we not rely on `requirements.txt` to supply the flake8 and pytest versions? Also, pre-commits rely on black and not flake8, I don't know if that will cause us to have different linting mechanisms

File: .github/workflows/python-app.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+    - name: Test with pytest
+      run: |
+        pytest
```
Comment (jonathantan1425) [inline .github/workflows/python-app.yml:53]: Will require changes to this line in the future to prepare the test database (changes are introduced in https://github.com/is212g7t5/spm/pull/9)

Comment (quinncheong) [reply .github/workflows/python-app.yml:53]: Hmm ok can you create a ticket and place in the backlog? I'm not too sure what to change it to.

File: .github/workflows/python-app.yml (hunk 2 above)
Comment (jonathantan1425) [reply .github/workflows/python-app.yml:23]: https://docs.github.com/en/actions/security-guides/encrypted-secrets should use this instead

File: .github/workflows/python-app.yml (hunk 4 above)
Comment (jonathantan1425) [reply .github/workflows/python-app.yml:53]: Created at https://is212g7t5.atlassian.net/browse/G7T5-46

**Review by deadline2359 — APPROVED** [review]

**Pull request merged**